### PR TITLE
feat(engine): implement ShadowRealm.prototype.importValue

### LINKED ISSUE #777: Implement ShadowRealm behind an explicit unsafe flag
## Summary

Implement the TC39 ShadowRealm proposal as an opt-in global built-in, exposed only when a new explicit flag is enabled.

Recommended flag name: `--unsafe-shadowrealm`, with matching config key `"unsafe-shadowrealm": true`.

The proposal specifies `ShadowRealm` as a global ECMAScript built-in constructor, not as a test262-only host hook. Because `ShadowRealm.prototype.evaluate()` performs dynamic source evaluation, GocciaScript should keep it unavailable by default and require a dedicated unsafe capability flag before installing `globalThis.ShadowRealm`.

## Why

The pinned built-ins test262 run currently has 64 ShadowRealm failures:

- `built-ins/ShadowRealm`: 64
- `built-ins/ShadowRealm/prototype`: 51
- `built-ins/ShadowRealm/prototype/evaluate`: 37
- exact message cluster: `ReferenceError: ShadowRealm is not defined`: 57

TC39 proposal status:

- Proposal: ShadowRealm
- Stage: 2.7
- Spec: ECMA-262 proposal
- test262 flags: `ShadowRealm`, `realms-tests`
- Proposal index SHA observed via TC39 MCP: `7510bde4071c3251114d1e7a244187c7c54da214`

The draft says `%ShadowRealm%` has global name `ShadowRealm`, and section 3.2 says the constructor is the initial value of the `"ShadowRealm"` property of the global object:
https://tc39.es/proposal-shadowrealm/

GocciaScript's project policy keeps dynamic code generation out of normal runtimes by default. `eval()` is private to the test262 host, and `Function()` already requires `--unsafe-function-constructor`. ShadowRealm should follow that explicit-capability pattern, but with its own flag so hosts can opt into this proposal independently.

## Current behavior

`ShadowRealm` is not installed.

test262 failures report `ReferenceError: ShadowRealm is not defined` for most ShadowRealm tests.

GocciaScript already has private realm/eval machinery for conformance:

- `$262.evalScript()`
- `$262.createRealm()`
- `GocciaScriptLoaderBare --test262-host`

Those hooks are not a public ShadowRealm implementation and should remain conformance host capabilities.

## Expected behavior

When `--unsafe-shadowrealm` is not enabled:

- `globalThis.ShadowRealm` is absent or unavailable, preserving the current no-dynamic-eval default.
- Normal runtime profiles do not gain a new source-evaluation capability.

When `--unsafe-shadowrealm` is enabled:

- `ShadowRealm` is installed as a global constructor.
- `new ShadowRealm()` creates a fresh realm with its own global object and intrinsics.
- `ShadowRealm()` without `new` throws `TypeError`.
- `ShadowRealm.prototype.evaluate(sourceText)` evaluates script source in the target ShadowRealm.
- Non-primitive, non-callable cross-realm results throw according to the proposal's callable-boundary rules.
- Callable results are exposed through wrapped function exotic behavior or the closest GocciaScript equivalent.
- `ShadowRealm.prototype[Symbol.toStringTag]` is `"ShadowRealm"`.
- test262 ShadowRealm and realms-tests cases can be enabled through the new flag.

## Scope notes

Likely implementation areas:

- CLI/config flag wiring near `--unsafe-function-constructor`
- `GocciaScriptLoaderBare` test262 host integration
- `TGocciaEngine.RegisterBuiltIns` global registration
- realm creation/evaluation machinery currently used by `$262.createRealm()` and `$262.evalScript()`
- test262 runner feature-flag handling for `ShadowRealm` / `realms-tests`

Non-goals for this issue:

- Do not expose ShadowRealm by default.
- Do not fold this into `--unsafe-function-constructor`; ShadowRealm needs a separate capability gate.
- Do not expose `$262` host hooks as public APIs.
- Do not implement broader Compartments or host virtualization APIs.
- Do not claim ShadowRealm is a security boundary; the proposal itself says it does not provide availability protection and is not a full confidentiality mechanism.

### PR DESCRIPTION
**Pull request by frostney**
## Summary

- Implements `ShadowRealm.prototype.importValue(specifier, exportName)` behind the existing `--unsafe-shadowrealm` gate, completing the method deferred from #795 (proposal-shadowrealm §3.4.2 → `RealmImportValue`).
- **Synchronous** (caller realm): `ValidateShadowRealmObject`, then `ToString(specifier)` — a throwing coercion (`valueOf`/`toString`/`@@toPrimitive`) or a `Symbol` propagates synchronously — then `exportName` must already be a `String` (type-checked, never coerced).
- **Asynchronous**: imports the module into the child realm, reads the requested export, and resolves the caller-realm promise with `GetWrappedValue` of the binding. A failed import (resolution / parse / link / evaluation throw), a missing export, or a non-wrappable export all reject with a **caller-realm `TypeError`**.
- The child realm resolves and reads modules the same way the realm that created it does: it shares the host module content provider, carries over the parent's module aliases (import map / `--alias`) and preprocessors, and resolves the specifier relative to the creating realm's entry path via a new read-only `TGocciaEngine.SourcePath` accessor. The module still evaluates against the child realm's own intrinsics. GocciaScript module loading is synchronous, so the import runs inline and the promise settles before `importValue` returns; the caller's `.then` reactions still run as microtasks.
- Follow-up to #795 (which closed #777); removes the `importValue` deferral recorded in ADR 0073.
- **Non-goal:** the 4 remaining `built-ins/ShadowRealm` test262 failures (`evaluate/**` ×3, `WrappedFunction/length.js` ×1) are the pre-existing-limitation edge cases #795 documented — untouched here.

## Testing

- [x] Verified no regressions and confirmed the new feature or bugfix in end-to-end JavaScript/TypeScript tests — full JS suite **10924/10924** in **both** interpreter and bytecode modes (adds `tests/built-ins/ShadowRealm/importValue.js`, 15 tests, plus a fixture module). test262 `built-ins/ShadowRealm/prototype/importValue` **12/12** in both modes with **0** wrapper-infra failures; the full `built-ins/ShadowRealm` subtree is now **60/64** (was 48/64). Adds a CI regression test in `scripts/test-cli-apps.ts` that runs `GocciaScriptLoader --alias … --unsafe-shadowrealm` and asserts the child realm resolves a host module alias.
- [x] Updated documentation — ADR 0073, `docs/built-ins.md`, `docs/language-tables.md`.
- [ ] Optional: Verified no regressions and confirmed the new feature or bugfix in native Pascal tests (if AST, scope, evaluator, or value types changed) — n/a; additive built-in method plus a read-only engine accessor, no AST/scope/evaluator/value-type changes.
- [ ] Optional: Verified no benchmark regressions or confirmed benchmark coverage for the change — n/a; no hot-path change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: source/units/Goccia.Builtins.GlobalShadowRealm.pas [hunk 1]
```diff
@@ -176,6 +183,13 @@ constructor TGocciaShadowRealmChildRealm.Create(
   FEngine.Preprocessors := AParentEngine.Preprocessors;
   FEngine.RefreshGlobalThis;
   EnableShadowRealm(FEngine);
+  // Let ShadowRealm.prototype.importValue resolve and read modules the same way
+  // the realm that created this one does. The imported module still evaluates
+  // in this child realm against its own intrinsics; only host file access is
+  // shared. Engine.Compatibility already propagates to the loader; preprocessors
+  // do not, so mirror them here too.
+  FEngine.ModuleLoader.SetContentProvider(AParentEngine.ContentProvider, False);
+  FEngine.ModuleLoader.Preprocessors := AParentEngine.Preprocessors;
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.Builtins.GlobalShadowRealm.pas:199]: Good catch — confirmed real and fixed.

Reproduced the divergence with `GocciaScriptLoader entry.js --unsafe-shadowrealm --alias aliased=./real.js`: the parent realm resolved `import("aliased")` (→ `99`), but `realm.importValue("aliased", "v")` rejected with a `TypeError` because the child kept a fresh, alias-less resolver.

Fix (child-realm setup in `Goccia.Builtins.GlobalShadowRealm.pas`) — carry over the creating realm's resolver aliases alongside the content provider/preprocessors:

```pascal
for AliasPair in AParentEngine.ModuleLoader.Resolver.Aliases do
  FEngine.ModuleLoader.Resolver.AddAlias(AliasPair.Key, AliasPair.Value);
```

After the fix the child resolves the alias identically (`child-resolved: 99`). Aliases are the only resolver state that can diverge here: `ConfigureModuleResolver` only ever populates aliases (import map + inline `--alias`), the resolver extensions default-match, and relative specifiers already resolve against the parent's entry path (`FEngine.SourcePath`). Added a CI regression test in `scripts/test-cli-apps.ts` that runs `GocciaScriptLoader --alias … --unsafe-shadowrealm` and asserts the child realm resolves the host alias.

**Comment by frostney** [comment]
@coderabbitai review
